aws/resource_aws_sagemaker_lifecycle_configuration.go

```diff
@@ -0,0 +1,168 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sagemaker"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsSagemakerLifeCycleConfiguration() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsSagemakerLifeCycleConfigurationCreate,
+		Read:   resourceAwsSagemakerLifeCycleConfigurationRead,
+		Update: resourceAwsSagemakerLifeCycleConfigurationUpdate,
+		Delete: resourceAwsSagemakerLifeCycleConfigurationDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"name": {
```

Comment:
Can you please add validation for this field? e.g.

```go
ValidateFunc: validation.StringLenBetween(1, 63),
```

Reply:
I'll use the func `validateSagemakerName` here to check also against the name pattern (https://github.com/awsdocs/amazon-sagemaker-developer-guide/blob/master/doc_source/API_CreateNotebookInstanceLifecycleConfig.md#request-parameters)